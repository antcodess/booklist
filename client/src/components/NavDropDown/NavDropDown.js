import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { selectCurrentUser } from '../../redux/user/user.selectors';
import { signOutStart } from '../../redux/user/user.actions';
import { toggleNavHidden } from '../../redux/nav/nav.actions';

import './NavDropDown.scss';

export const NavDropDown = ({ history, currentUser, signOutStart, toggleNavHidden }) => {
  return (
    <div className='nav-dropdown'>
      <ul className='nav-options'>
        <li>
          <span className='nav-option' id='nav-dropdown-shop-link' onClick={() => {
            history.push('/shop')
            toggleNavHidden()
          }}>
            Shop
          </span>
          <div className='nav-border'></div>
        </li>
        {
          currentUser ? 
          <React.Fragment>
            <li>
              <span className='nav-option' id='nav-dropdown-account-link'onClick={() => {
                history.push('/account')
                toggleNavHidden()
              }}>
                Account
              </span>
              <div className='nav-border'></div>
            </li>
            <li>
              <span className='nav-option' id='nav-dropdown-signout-link' onClick={() => {
                signOutStart()
                toggleNavHidden()
              }}>
                SignOut
              </span>
              <div className='nav-border'></div>
            </li>
          </React.Fragment>
          : 
          <li>
            <span className='nav-option' id='nav-dropdown-signin-link' onClick={() => {
              history.push('/signin')
              toggleNavHidden()
            }}>
              SignIn
            </span>
            <div className='nav-border'></div>
          </li>
        }
        <li>
          <span className='nav-option checkout' id='nav-dropdown-checkout-link' onClick={() => {
            history.push('/checkout')
          }}>
            Checkout
          </span>
        </li>
      </ul>
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  currentUser: selectCurrentUser
});

export const mapDispatchToProps = dispatch => ({
  signOutStart: () => dispatch(signOutStart()),
  toggleNavHidden: () => dispatch(toggleNavHidden())
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(NavDropDown));