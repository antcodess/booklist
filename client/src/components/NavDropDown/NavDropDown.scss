.nav-dropdown {
  background-color: white;
  position: absolute;
  width: 100px;
  display: flex;
  flex-direction: column;
  border-radius: 5px;
  box-shadow: 2px 1px 10px #c3c3c3;
  top: 52px;
  left: 176px;
  z-index: 5;

  .nav-options {
    margin-top: 10px;
    padding-bottom: 10px;
    
    li {
      margin-top: 5px;
      padding-bottom: 5px;

      .nav-border {
        padding-top: 8px;
        width: 5%;
        border-bottom: 1px #D9D9D9 solid;
        margin: auto;
      }
    }

    .nav-option { 
      font-family: "Cera GR Bold";
      font-size: 1.125rem;

      &:hover {
        color: #0080DE;
      }
      
      .cart-icon {
        display: flex;
        margin: auto;
      }

      .shopping-icon {
        background-size: 44px;
      }
    }
  }
}

@media only screen and (max-width: 500px) {
  .nav-dropdown {
    /* left: 198px; */
    left: 175px;
  }
}

@media only screen and (max-width: 480px) {
  .nav-dropdown {
    left: 176px;
  } 
}

@media only screen and (max-width: 400px) {
  .nav-dropdown {
    top: 42px;
    left: 130px;
    z-index: 5;
  }
}