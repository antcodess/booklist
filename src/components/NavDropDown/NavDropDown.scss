.nav-dropdown {
  background-color: white;
  position: absolute;
  width: 100px;
  display: flex;
  flex-direction: column;
  border-radius: 5px;
  box-shadow: 2px 1px 10px #c3c3c3;
  top: 52px;
  left: 176px;
  z-index: 5;

  .nav-options {
    margin-top: 10px;
    padding-bottom: 10px;
    
    li {
      margin-top: 5px;
      padding-bottom: 5px;

      .nav-border {
        padding-top: 8px;
        width: 5%;
        border-bottom: 1px #D9D9D9 solid;
        margin: auto;
      }
    }

    .nav-option { 
      font-family: "Cera GR Bold";
      font-size: 1.125rem;
      
      .cart-icon {
        display: flex;
        margin: auto;
      }

      .shopping-icon {
        background-size: 44px;
      }
    }
  }
}