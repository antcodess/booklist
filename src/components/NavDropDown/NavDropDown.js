import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { selectCurrentUser } from '../../redux/user/user.selectors';
import { signOutStart } from '../../redux/user/user.actions';
import { toggleNavHidden } from '../../redux/nav/nav.actions';
import './NavDropDown.scss';

const NavDropDown = ({ history, currentUser, signOutStart, toggleNavHidden }) => {
  return (
    <div className='nav-dropdown'>
      <ul className='nav-options'>
        <li>
          <span className='nav-option' onClick={() => {
            history.push('/shop')
            toggleNavHidden()
          }}>
            Shop
          </span>
          <div className='nav-border'></div>
        </li>
        {
          currentUser ? 
          <React.Fragment>
            <li>
              <span className='nav-option' onClick={() => {
                history.push('/account')
                toggleNavHidden()
              }}>
                Account
              </span>
              <div className='nav-border'></div>
            </li>
            <li>
              <span className='nav-option' onClick={() => {
                signOutStart()
                toggleNavHidden()
              }}>
                SignOut
              </span>
              <div className='nav-border'></div>
            </li>
          </React.Fragment>
          : 
          <li>
            <span className='nav-option' onClick={() => {
              history.push('/signin')
              toggleNavHidden()
            }}>
              SignIn
            </span>
            <div className='nav-border'></div>
          </li>
        }
        <li>
          <span className='nav-option checkout' onClick={() => {
            history.push('/checkout')
            toggleNavHidden()
          }}>
            Checkout
          </span>
        </li>
      </ul>
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  currentUser: selectCurrentUser
});

const mapDispatchToProps = dispatch => ({
  signOutStart: () => dispatch(signOutStart()),
  toggleNavHidden: () => dispatch(toggleNavHidden())
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(NavDropDown));