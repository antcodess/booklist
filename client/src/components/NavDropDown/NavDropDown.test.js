import React from 'react';
import { shallow } from 'enzyme';
import { NavDropDown } from './NavDropDown';
import { mapDispatchToProps } from './NavDropDown';

describe('NavDropDown Component', () => {
  let wrapper;
  let wrapper2;
  let mockHistory;
  let mockCurrentUser; 
  let mockSignOutStart; 
  let mockToggleNavHidden;
  let dummyFunction = jest.fn();
  
  beforeEach(() => {

    mockHistory = { push: jest.fn() };
    mockCurrentUser = { id: 1, name: 'Sample User'};
    mockSignOutStart = jest.fn();
    mockToggleNavHidden = jest.fn();

    const mockProps = {
      history: mockHistory, 
      currentUser: mockCurrentUser, 
      signOutStart: mockSignOutStart, 
      toggleNavHidden: mockToggleNavHidden
    };

    const mockProps2 = {
      history: mockHistory,
      currentUser: null, 
      signOutStart: mockSignOutStart, 
      toggleNavHidden: mockToggleNavHidden
    };
    
    // Signed In User
    wrapper = shallow(<NavDropDown { ...mockProps } />);
    // Signed Out User
    wrapper2 = shallow(<NavDropDown { ...mockProps2 } />);
  });

  it('should render NavDropDown Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push when clicked for nav-dropdown-shop-link div', () => {
    wrapper.find('#nav-dropdown-shop-link').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it('should call toggleCartHidden when clicked for nav-dropdown-shop-link div', () => {
    wrapper.find('#nav-dropdown-shop-link').simulate('click');
    expect(mockToggleNavHidden).toHaveBeenCalled();
  });

  it('should call history.push when clicked for nav-dropdown-account-link div', () => {
    wrapper.find('#nav-dropdown-account-link').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it('should call toggleCartHidden when clicked for nav-dropdown-account-link div', () => {
    wrapper.find('#nav-dropdown-account-link').simulate('click');
    expect(mockToggleNavHidden).toHaveBeenCalled();
  });

  it('should call signOutStart when clicked for nav-dropdown-signout-link div', () => {
    wrapper.find('#nav-dropdown-signout-link').simulate('click');
    expect(mockSignOutStart).toHaveBeenCalled();
  });

  it('should call toggleCartHidden when clicked for nav-dropdown-signout-link div', () => {
    wrapper.find('#nav-dropdown-signout-link').simulate('click');
    expect(mockToggleNavHidden).toHaveBeenCalled();
  });

  it('should call history.push when clicked for nav-dropdown-signin-link div', () => {
    wrapper2.find('#nav-dropdown-signin-link').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it('should call toggleCartHidden when clicked for nav-dropdown-signin-link div', () => {
    wrapper2.find('#nav-dropdown-signin-link').simulate('click');
    expect(mockToggleNavHidden).toHaveBeenCalled();
  });

  it('should call history.push when clicked for nav-dropdown-checkout-link div', () => {
    wrapper.find('#nav-dropdown-checkout-link').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it('should show account link if user is signed in', () => {
    expect(wrapper.exists('#nav-dropdown-account-link')).toBe(true);
  });

  it('should not show account link if user is not signed in', () => {
    expect(wrapper2.exists('#nav-dropdown-account-link')).toBe(false);
  });

  it('should show signout link if user is signed in', () => {
    expect(wrapper.exists('#nav-dropdown-signout-link')).toBe(true);
  });

  it('should not show signout link if user is not signed in', () => {
    expect(wrapper2.exists('#nav-dropdown-signout-link')).toBe(false);
  });

  it('should not show signin link if user is signed in', () => {
    expect(wrapper.exists('#nav-dropdown-signin-link')).toBe(false);
  });

  it('should show signin link if user is not signed in', () => {
    expect(wrapper2.exists('#nav-dropdown-signin-link')).toBe(true);
  });

  it('should dispatch signOutStart action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).signOutStart();
    expect(dispatch.mock.calls[0][0]).toEqual({"type": "SIGN_OUT_START"});
  });

  it('should dispatch toggleNavHidden action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).toggleNavHidden();
    expect(dispatch.mock.calls[0][0]).toEqual({"type": "TOGGLE_NAV_HIDDEN"});
  });
});
