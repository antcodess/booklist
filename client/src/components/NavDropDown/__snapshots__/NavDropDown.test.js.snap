// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavDropDown Component should render NavDropDown Component 1`] = `ShallowWrapper {}`;
